### DISCUSSION TITLE
Unicode NFC file names

### DISCUSSION THREAD

**Discussion by ManfredXG** [mikekazakov/nimble-commander#321] [Ideas]
Hi, I'm new here ...
Many many tanks for this piece of software.

[ Hint: NFD stores ü as 0075 0308, NFC stores it as 00FC ]

I have an idea to improve it (in my view):
When you create/rename folders/files, Nimble Commander always uses NFD-names (like Mac Finder).
The MacOS-Terminal uses NFC or NFD, depending on what you type.
I know, MacOS is storing both, but ls-command and C-readdir()-function deliver the primary name-form.
My suggestion:
Either like Terminal or to have a switch (Always NFD, always NFC, as typed in)
What's your opinion to this?

Manfred

**Comment by mikekazakov** [comment]
Hi, you're welcome!

IFAIK filesystem paths should be always decomposed on MacOSX:
- https://developer.apple.com/library/archive/technotes/tn/tn1150.html#UnicodeSubtleties
- https://github.com/apple/swift-corelibs-foundation/blob/ca3669eb9ac282c649e71824d9357dbe140c8251/CoreFoundation/String.subproj/CFStringEncodings.c#L1011
- https://web.archive.org/web/20140812023313/http://developer.apple.com/library/ios/documentation/MacOSX/Conceptual/BPInternational/Articles/FileEncodings.html
- https://developer.apple.com/support/downloads/Apple-File-System-Reference.pdf
etc...

Why do you care about the normalization method in the first place?

**Comment by ManfredXG** [comment]
Hi,
after calling C-function readdir() my program has to convert from NFD to NFC.
That would then be unnecessary.
Manfred

**Comment by mikekazakov** [comment]
And what this has to do with either MacOSX or Nimble Commander?
If your code needs a Unicode string in a specific normal form - this normalizations has to be performed by that program.

**Reply by ManfredXG** [reply]
From your point of view, you're right.
But I think, it would be a great feature.
Mac-Terminal has this feature and I like it.
But it's not "vital" for me, only nice to have ...
